File: main/client/src/mill/main/client/lock/MemoryTryLocked.java
package mill.main.client.lock;

class MemoryTryLocked extends MemoryLocked implements TryLocked {
  public MemoryTryLocked(java.util.concurrent.locks.Lock lock) {
    super(lock);
  }

  public boolean isLocked() {
    return lock != null;
  }

  public void release() throws Exception {
    if (isLocked()) super.release();
  }
}

File: main/client/src/mill/main/client/lock/MemoryTryLocked.scala
package mill.main.client.lock

import java.util.concurrent.locks.Lock

class MemoryTryLocked(lock: Lock) extends MemoryLocked(lock) with TryLocked {

  def isLocked: Boolean = lock != null

  @throws[Exception]
  override def release(): Unit = {
    if (isLocked) super.release()
  }
}
